src/unicode-utils-posix.cc

```diff
@@ -41,8 +41,11 @@ int UnicodeUtils::bytes_in_characters(const char* string, int characters) {
       break;
 
     bytes += characterLength;
-    if (--characters == 0)
-      break;
+    if (characterLength == 4)
+      characters -= 2;
```

Comment:
Non-BMP characters are always encoded with 4 bytes in UTF-8 and 2 code units (surrogate pairs) in UTF-16. BMP characters are always 1 to 3 bytes in UTF-8 and 1 code unit in UTF-16.

UTF-8 doesn’t officially support encoding characters using 5 or more bytes. Although it’s technically possible, there will never be Unicode characters above U+10FFFF (which would require 5+ bytes in UTF-8) due to the fact that it’s not possible to encode them in UTF-16.